tests/system/src/test/java/test/com/sun/javafx/animation/SynchronizationTest.java

```diff
@@ -0,0 +1,104 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.com.sun.javafx.animation;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.stage.Stage;
+import test.util.Util;
+
+// Based on https://bugs.openjdk.org/browse/JDK-8159048
+public class SynchronizationTest {
+
+    private static final CountDownLatch startupLatch = new CountDownLatch(1);
+    private static Stage primaryStage;
+
+    public static class TestApp extends Application {
+
+        @Override
+        public void start(Stage stage) throws Exception {
+            primaryStage = stage;
+            startupLatch.countDown();
+        }
+    }
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        Util.launch(startupLatch, TestApp.class);
+    }
+
+    @AfterAll
+    public static void shutdown() {
+        Util.shutdown(primaryStage);
+    }
+
+    /**
+     * Number of seconds to wait for a failure. If an exception is not thrown in this time, it's assumed it won't be
+     * thrown later too.
+     */
+    private static final int GRACE_PERIOD = 15;
+
+    final private AtomicBoolean failed = new AtomicBoolean(false);
+    final private CountDownLatch waiter = new CountDownLatch(1);
+    final private ExecutorService executor = Executors.newCachedThreadPool();
+
+    private Thread thread;
+    private Throwable throwable;
+
+    protected void runTest(Runnable runnable) throws InterruptedException {
+        Platform.runLater(() -> registerExceptionHandler());
+
+        for (int i = 0; i < 10; i++) {
+            executor.submit(runnable);
+        }
```

Comment:
I did a quick prototype of the idea I mentioned in my last comment, and it looks like it works:

```
        Runnable wrappedRunnable = () -> {
            try {
                runnable.run();
            } catch (Throwable e) {
                thread = Thread.currentThread();
                throwable = e;
                failed.set(true);
                waiter.countDown();
            }
        };

        for (int i = 0; i < 10; i++) {
            executor.submit(wrappedRunnable);
        }
```

This obviates the need for the `registerExceptionHandler` calls in the two tests themselves (this one in _this_ method is still needed).

Reply:
I don't think the executor service catches the exception because if it did then the `AnimationTimer` test would also always pass. I do get the AIIOB exceptions from the background thread caught in the handler (which fails the test). This test is rather stable for me. The `Animation` one is more problematic, sometimes failing and sometimes not when it should fail.

Reply:
Based on the testing I did, the executor service does prevent at least some exceptions that happen on the test thread from reaching the uncaught exception handler. An explicit try/catch will catch those.

Reply:
Alright, will implement the change.